tests/unit-tests/did_doc_resolve_service_test.go

```diff
@@ -0,0 +1,145 @@
+package tests
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	didTypes "github.com/cheqd/cheqd-node/api/v2/cheqd/did/v2"
+	resourceTypes "github.com/cheqd/cheqd-node/api/v2/cheqd/resource/v2"
+	"github.com/cheqd/did-resolver/services"
+	"github.com/cheqd/did-resolver/types"
+)
+
+type resolveTestCase struct {
+	ledgerService          MockLedgerService
+	resolutionType         types.ContentType
+	identifier             string
+	method                 string
+	namespace              string
+	expectedDID            *types.DidDoc
+	expectedMetadata       types.ResolutionDidDocMetadata
+	expectedResolutionType types.ContentType
+	expectedError          *types.IdentityError
+}
+
+var _ = DescribeTable("Test Resolve method", func(testCase resolveTestCase) {
+	did := fmt.Sprintf("did:%s:%s:%s", testCase.method, testCase.namespace, testCase.identifier)
+
+	diddocService := services.NewDIDDocService("cheqd", testCase.ledgerService)
+	expectedDIDProperties := types.DidProperties{
+		DidString:        did,
+		MethodSpecificId: testCase.identifier,
+		Method:           testCase.method,
+	}
+
+	if (testCase.resolutionType == "" || testCase.resolutionType == types.DIDJSONLD) && testCase.expectedError == nil {
+		testCase.expectedDID.Context = []string{types.DIDSchemaJSONLD, types.JsonWebKey2020JSONLD}
+	} else if testCase.expectedDID != nil {
+		testCase.expectedDID.Context = nil
+	}
+
+	expectedContentType := testCase.expectedResolutionType
+	if expectedContentType == "" {
+		expectedContentType = testCase.resolutionType
+	}
+
+	resolutionResult, err := diddocService.Resolve(did, "", testCase.resolutionType)
+	if testCase.expectedError != nil {
+		Expect(testCase.expectedError.Code).To(Equal(err.Code))
+		Expect(testCase.expectedError.Message).To(Equal(err.Message))
+	} else {
+		Expect(err).To(BeNil())
+		Expect(testCase.expectedDID).To(Equal(resolutionResult.Did))
+		Expect(testCase.expectedMetadata).To(Equal(resolutionResult.Metadata))
+		Expect(expectedContentType).To(Equal(resolutionResult.ResolutionMetadata.ContentType))
+		Expect(expectedDIDProperties).To(Equal(resolutionResult.ResolutionMetadata.DidProperties))
+	}
+},
+
+	Entry(
+		"Successful resolution",
+		resolveTestCase{
+			ledgerService:    NewMockLedgerService(&validDIDDoc, &validMetadata, &validResource),
```

Comment:
Maybe we need to simplify it somehow here? From my point of view it's hard to understand what the actual output is expected. Let's maybe make here at least the next changes:
1. Let's create MockLedgerService once and with all of variants of DIDDoc and resources. I think that test case only should include the URI as input and expected Go structure as output.
2. Let's use actual data formats  for each response.

Reply:
Done! ✅